corehq/motech/repeaters/views/repeat_records.py

```diff
@@ -217,27 +244,41 @@ def headers(self):
             DataTablesColumn(_('Cancel or Requeue payload'))
         ]
         if toggles.SUPPORT.enabled_for_request(self.request):
-            columns.insert(1, DataTablesColumn(_('Payload ID')))
+            columns.insert(2, DataTablesColumn(_('Payload ID')))
 
         return DataTablesHeader(*columns)
 
     @property
     def report_context(self):
         context = super(DomainForwardingRepeatRecords, self).report_context
+
+        total = get_repeat_record_count(self.domain)
```

Comment:
If you change these three lines, and make `self.repeater_id` a property, then the totals will change when a Repeater is selected. See the diff below.

If I understand correctly (and please correct me if I'm wrong), corehq.apps.data_interfaces.utils._get_ids() will only return the payload IDs for the Repeater that is selected. Is that right?

If so, this change would be more accurate, and the behaviour that the user would expect when they select a Repeater.

```
diff --git a/corehq/motech/repeaters/views/repeat_records.py b/corehq/motech/repeaters/views/repeat_records.py
index 4cbb581397e..6b38f0f7ab1 100644
--- a/corehq/motech/repeaters/views/repeat_records.py
+++ b/corehq/motech/repeaters/views/repeat_records.py
@@ -166,9 +166,12 @@ class DomainForwardingRepeatRecords(GenericTabularReport):
     def payload_id(self):
         return self.request.GET.get('payload_id', None)
 
+    @property
+    def repeater_id(self):
+        return self.request.GET.get('repeater', None)
+
     @property
     def rows(self):
-        self.repeater_id = self.request.GET.get('repeater', None)
         self.state = self.request.GET.get('record_state', None)
         if self.payload_id:
             end = self.pagination.start + self.pagination.count
@@ -252,9 +255,9 @@ class DomainForwardingRepeatRecords(GenericTabularReport):
     def report_context(self):
         context = super(DomainForwardingRepeatRecords, self).report_context
 
-        total = get_repeat_record_count(self.domain)
-        total_cancel = get_pending_repeat_record_count(self.domain, None)
-        total_requeue = get_cancelled_repeat_record_count(self.domain, None)
+        total = get_repeat_record_count(self.domain, self.repeater_id)
+        total_cancel = get_pending_repeat_record_count(self.domain, self.repeater_id)
+        total_requeue = get_cancelled_repeat_record_count(self.domain, self.repeater_id)
 
         form_query_string = self.request.GET.urlencode()
         form_query_string_requeue = _change_record_state(form_query_string, 'CANCELLED')
```